src/test/ui/resolve/typo-suggestion-for-variable-with-name-similar-to-struct-field.stderr

```diff
@@ -0,0 +1,36 @@
+error[E0425]: cannot find value `config` in this scope
+  --> $DIR/typo-suggestion-for-variable-with-name-similar-to-struct-field.rs:7:16
+   |
+LL |         Self { config }
+   |                ^^^^^^
+   |                |
+   |                a field by this name exists in `Self`
+   |                help: a local variable with a similar name exists: `cofig`
+
+error[E0425]: cannot find value `config` in this scope
+  --> $DIR/typo-suggestion-for-variable-with-name-similar-to-struct-field.rs:11:20
+   |
+LL |         println!("{config}");
+   |                    ^^^^^^
+   |                    |
+   |                    a field by this name exists in `Self`
+   |                    help: a local variable with a similar name exists: `cofig`
+
+error[E0425]: cannot find value `config` in this scope
+  --> $DIR/typo-suggestion-for-variable-with-name-similar-to-struct-field.rs:15:20
+   |
+LL |         println!("{config}");
+   |                    ^^^^^^
+   |
+help: you might have meant to use the available field
+   |
+LL |         println!("{self.config}");
```

Comment:
Drive-by comment: this probably shouldn't be emitted, as it won't compile.

Reply:
Yeah, I think that's a bit orthogonal to this diff, though. It might be possible to detect this case from HIR though. 

@TaKO8Ki, I would appreciate if you find some time to investigate suppressing this. Don't block working on this if it's more difficult than it seems -- if not immediately easy to fix, then could you file an issue and downgrade the suggestion from `MachineApplicable` to `MaybeIncorrect`? :sweat_smile:

Reply:
> I would appreciate if you find some time to investigate suppressing this. Don't block this PR on that if it's more difficult than it seems -- if not immediately easy to fix, then could you file an issue and downgrade the suggestion from MachineApplicable to MaybeIncorrect? 😅

Ok. I will investigate it.

Reply:
I opened #97311. Can I work on this problem in a new PR?

Reply:
Yeah, sounds good.